### PR TITLE
CORDA-1647: Break circular dependency on publish-utils

### PR DESCRIPTION
**Pull request by Clintonio**
This PR essentially decomposes the logic of publish utils into the root build.gradle. This is the better, more gradle-y way of doing this. We have been mis-using gradle build files for some time ignoring the power of the ability to configure a subset of projects. This is all the publish utils plugins does.

This PR will lead way to remove publish utils from this repo in leiu of a single solution within the main build.gradle or buildSrc. 

Also;

* Added new gradle properties flags (-D) to configure the bintray details. 
* Added new gradle properties flags (-D) to configure the artifactory details.

Also:

* Removed automatic publishing from the gradle plugin plugin
* Upgraded to gradle 4.8 to fix an issue with the pom configuration (it was a hack before)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -51,52 +46,102 @@ allprojects {
...
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    bintray {
+        user = System.getenv('CORDA_BINTRAY_USER') || System.getProperty('corda.bintray.user')
+        key = System.getenv('CORDA_BINTRAY_KEY') || System.getPropert('corda.bintray.key')
```
Comment (chrisr3) [inline build.gradle]: Ooopsee!! Typo! I think you meant
```groovy
System.getProperty('corda.bintray.key')
```

File: build.gradle [hunk 2]
```diff
@@ -51,52 +46,102 @@ allprojects {
...
-    gpgPassphrase = System.getenv('CORDA_BINTRAY_GPG_PASSPHRASE')
-    publications = ['cordformation', 'quasar-utils', 'cordform-common', 'api-scanner', 'cordapp']
-    license {
-        name = 'Apache-2.0'
-        url = 'https://www.apache.org/licenses/LICENSE-2.0'
-        distribution = 'repo'
+def publishProjects = project.subprojects - project('api-scanner:annotations')
+
+configure(publishProjects) { subproject ->
+    apply plugin: 'java'
```
Comment (chrisr3) [inline build.gradle]: I'm slightly concerned that you're always applying the `java` plugin, even for plugins written in Kotlin or Groovy, but I suppose it makes no material difference to how everything is compiled.

File: build.gradle (hunk 1 above)
Comment (Clintonio) [reply build.gradle]: I clearly rushed this before I left...

File: build.gradle (hunk 2 above)
Comment (Clintonio) [reply build.gradle]: Yeah, we do it in the main corda repo too. Maybe we can fine grain it later but this was needed to get the sourceJar and docsJar tasks working.

File: build.gradle [hunk 3]
```diff
@@ -26,14 +23,12 @@ buildscript {
     }
 
     dependencies {
-        classpath "net.corda.plugins:publish-utils:$gradle_plugins_version"
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.7.3'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         classpath "org.jfrog.buildinfo:build-info-extractor-gradle:$artifactory_plugin_version"
```
[the remark this answers is not in the document]
Comment (Clintonio) [reply build.gradle]: Updated. Also removed constants.properties since it existed because of the hack this PR fixes. Yay.

File: build.gradle [hunk 4]
```diff
@@ -51,52 +41,102 @@ allprojects {
...
+                gpg {
+                    sign = true
+                    passphrase = System.getenv('CORDA_BINTRAY_GPG_PASSPHRASE')
+                }
+            }
+        }
     }
+
+    task install(dependsOn: 'publishToMavenLocal')
 }
```
Comment (chrisr3) [inline build.gradle]: Move this task underneath the `sourceJar` and `javadocJar` tasks.

**Comment by Clintonio** [comment]
I'll update TC when this PR is approved. That way there's nearly 0 time where the build logic differs from the code.

File: build.gradle [hunk 5]
```diff
@@ -79,24 +124,19 @@ artifactory {
...
+            if (publishProjects.contains(project)) {
+                publications(project.name)
             }
         }
     }
 }
 
 wrapper {
-    gradleVersion = "4.4.1"
+    gradleVersion = "4.8"
```
Comment (chrisr3) [inline build.gradle]: _No, **not this**!_ We want [Gradle 4.8.1](https://github.com/gradle/gradle/releases/tag/v4.8.1)!!!!!!! :smiling_imp:

Comment (Clintonio) [reply build.gradle]: Hah, perfectly bad timing :D

**Comment by chrisr3** [comment]
I'm not convinced that the wrapper JAR has changed between Gradle 4.8 and Gradle 4.8.1, but anyway...

**Pull request merged**